### PR TITLE
dlog: Fix issue where the fast-logging broke backward compat

### PR DESCRIPTION
**Pull request by LukeShu**
> This PR is on top of https://github.com/datawire/dlib/pull/34, go merge that first.

https://github.com/datawire/dlib/pull/28 changed the `Logger` interface, breaking any existing implementors of the interface.  That's not OK.

So revert that interface (and add a comment about how it's frozen), and add an opt-in `OptimizedLogger` interface for Jose's optimizations.

### CODE REVIEW

File: dlog/logger.go [hunk 1]
```diff
@@ -47,13 +50,38 @@ type Logger interface {
 	StdLogger(LogLevel) *log.Logger
 
 	// Log actually logs a message.
-	Log(level LogLevel, args ...interface{})
+	Log(level LogLevel, msg string)
```
Comment (thallgren) [inline dlog/logger.go:53]: Reverting this will break Telepresence, which I assume is OK, given that we're not using a released version.

**Review by thallgren — COMMENTED** [review]
Since I consider the non-optimized logger a design flaw, I'd prefer if this PR wasn't merged and the version instead bumped to 2.0.0. That way the backward compatibility wouldn't be broken for 1.x users and the flaw could still be removed.

**Review by josecv — APPROVED** [review]
this lgtm from a code perspective, i do kind of agree we might wanna release a 2.0.0 but im okay with unbreaking the 1.x line for now

**Comment by LukeShu** [comment]
> Since I consider the non-optimized logger a design flaw

I don't think I agree. I think keeping the required-interface minimal and letting dlog handle as much complexity as it can is a good thing.

**Comment by thallgren** [comment]
A 2.0 version would be good for three reasons.
1. It wouldn't require this somewhat cludgy solution with casts everywhere.
2. It would give the 1.2 implementers a chance to simply stay put while at the same time give them an incentive to improve their implementation.
3. It would give Telepresence a version that doesn't break us.

My motivation for calling the old version flawed is that unnecessary string manipulation is expensive and the added complexity is very trivial. In fact, in most cases it's just delegated to another log implementation.

**Pull request merged**